src/decorator/tree/Tree.ts

```diff
@@ -1,19 +1,23 @@
 import {getMetadataArgsStorage} from "../../";
 import {TreeMetadataArgs} from "../../metadata-args/TreeMetadataArgs";
 import {TreeType} from "../../metadata/types/TreeTypes";
+import {ClosureTreeOptions} from "../../metadata/types/ClosureTreeOptions";
 
 /**
  * Marks entity to work like a tree.
  * Tree pattern that will be used for the tree entity should be specified.
  * @TreeParent decorator must be used in tree entities.
  * TreeRepository can be used to manipulate with tree entities.
  */
-export function Tree(type: TreeType): ClassDecorator {
+export function Tree(type: TreeType, options?: ClosureTreeOptions): ClassDecorator {
     return function (target: Function) {
-
-        getMetadataArgsStorage().trees.push({
+        let args: TreeMetadataArgs = {
             target: target,
             type: type
-        } as TreeMetadataArgs);
+        };
+        if (type === "closure-table") {
+            args.options = options;
```

Comment:
Stay with the old code but add `options: type === "closure-table" ? options : undefined`

Reply:
done